lib/fdoprot.c

```diff
@@ -191,20 +191,26 @@ bool fdo_process_states(fdo_prot_t *ps)
 		}
 
 		if (ps->state != FDO_STATE_DONE && state_fn && state_fn(ps)) {
-			char err_msg[64];
+			char err_msg[64] = {0};
+			size_t err_msg_sz = 0;
 
 			(void)snprintf_s_i(err_msg, sizeof(err_msg),
 					   "msg%d: message parse error",
 					   ps->state);
-			ps->state = FDO_STATE_ERROR;
+			err_msg_sz = strnlen_s(err_msg, sizeof(err_msg));
+			if (!err_msg_sz || err_msg_sz == sizeof(err_msg)) {
```

Comment:
I don't understand the logic here - if err_msg_sz is equal to the sizeof(err_msg) then why would it be an error?

Reply:
sizeof(err_msg) equals 64, that is the maximum size of the buffer.

err_msg_sz is the size of the NULL-terminate string contained within that buffer. If strnlen_s() returns the full size of buffer, it means it didn't find a NULL character in the provided buffer.